### ISSUE TITLE
Sign > Share > Donate: Add "Other" amount button to donate buttons

### ISSUE THREAD

**Issue by rachelhbd** [350org/ak_intl_v3#382]
We would like to add an "Other" amount button to the donation step, eg. step 2 here: https://act.350.org/sign/sukhada-demo-share-page

![Image](https://user-images.githubusercontent.com/22962348/188920399-ac5f2239-1f6f-4d9b-9b56-557a116b5f32.png)

which wouldn't have an amount defined and would (if possible) preselect the "other" amount box on the donate form eg.

![Image](https://user-images.githubusercontent.com/22962348/188919626-c272e717-bc5c-44a2-b348-af84eb241d2e.png)

Don't worry about the styling of this as will address that in another ticket (https://github.com/350org/ak_intl_v3/issues/383) - for now I think we could just have the text say "Other" in the same font, and I'd leave font size and button width up to your judgment

**Comment by rachelhbd** [comment]
@sukhada I marked this as "Ready for dev" but if you have any questions let me know!

**Comment by sukhada** [comment]
@rachelhbd do we want an "Other" button in addition to the existing buttons, or for the button to replace one of the existing donate button options?

**Comment by rachelhbd** [comment]
In addition, please!

**Comment by sukhada** [comment]
This is ready to test via the Sept '22 templateset: https://act.350.org/admin/cms/templateset/460/templates

**Comment by rachelhbd** [comment]
Thanks @sukhada! This looks good to me when I select the United States, but if I select the United Kingdom or France I get a "NAN" message on the button which doesn't seem quite right:

![Image](https://user-images.githubusercontent.com/22962348/192336956-6077debf-61a8-44f8-b917-83720067af38.png)
![Image](https://user-images.githubusercontent.com/22962348/192336954-897c7326-5cc0-46eb-9e91-c7722cb3a331.png)
![Image](https://user-images.githubusercontent.com/22962348/192336955-bd6f60af-343b-4c80-b4c0-f0e3b6dd41ec.png)

**Comment by rachelhbd** [comment]
Also on mobile, seems like the line of buttons are too wide on some mobile devices. 

![Image](https://user-images.githubusercontent.com/22962348/192342376-d07f11e3-4352-433b-b944-95b08bd7cb3d.png)


On mobile, what you think is the best layout? 
- We could have 2 buttons on each line (though if in some cases the translation of "Other" is more characters, we might need to reduce the text size dynamically to keep the button sizes?)
- Or we could stack the buttons vertically
- Put the Other button on it's own line, make it wide, like we do for the $350 amount on the banner eg.
![Image](https://user-images.githubusercontent.com/22962348/192344438-75c924a4-fe92-436a-b33d-a6f4b45374ea.png)

**Comment by rachelhbd** [comment]
Ooh that's useful feedback thanks! I think with the alignment slightly off as well it looks a tad distracting 🤔 

I'm wondering if it makes sense to switch tactics to 6 buttons here for both mobile and desktop, using the same button amounts as on this donate page: https://act.350.org/donate/qa-apr-22-em-us-earth-month/ (3.50, 25, 50, 100, 350, other) -- this is something Bianca had requested before but I hadn't prioritised yet. But if it makes the mobile layout an easier problem to solve as well, maybe it's worth bringing forward and adding to the scope?

If that doesn't seem feasible/a good idea in the next couple of days, then let's keep it as is and we can revisit button designs/layouts here in the next sprint!

Either way please can we wrap the "Other" button text in a language string eg.
`{% filter ak_text:"afteraction_donate_other_amount" %}Other{% endfilter %}`?

**Comment by sukhada** [comment]
@rachelhbd Here's how that looks on desktop + mobile:

![Screen Shot 2022-09-27 at 1 25 01 PM](https://user-images.githubusercontent.com/1696078/192628237-f08b2f01-11dd-4e47-8f60-9321d42b7225.png)
![Screen Shot 2022-09-27 at 1 25 06 PM](https://user-images.githubusercontent.com/1696078/192628246-3008e750-40bd-4383-b0fb-56679524b5b9.png)

I also updated the templateset, so you can check it out and see if it looks good (also added the language string for "Other" as well).

**Comment by rachelhbd** [comment]
Thanks a lot @sukhada! I'm going to mark this one Done in that we've successfully added the "Other" button, but I'm realising there are more UX things we need to iron out (button amounts, colours, layout) before we can go live with this. Sorry, I didn't think through the implications of this request for the mobile layout before!

Let's keep this out of the September release for now, but I'll use what we have here to get a round of feedback, and bring an updated request for the October or November sprint, including a design preview if possible!
